pkg/services/authz/rbac/mapper.go

```diff
@@ -63,6 +63,15 @@ func newMapper() mapper {
 		"iam.grafana.app": {
 			"teams": newResourceTranslation("teams", "id", false),
 		},
+		"query.grafana.app": {
+			"query": translation{
+				resource:  "query",
```

Comment:
To build the scope I think this should be `datasources`
```suggestion
				resource:  "datasources",
```

Reply:
@gamab this is interesting. actually, in this case, i am hardcoding the at https://github.com/grafana/grafana/blob/68b55a8243e8a0f99faadff6b2a298c85eff7e6d/pkg/services/authz/rbac/mapper.go#L71 , so it's not calculate based on this string.

also, it works both with my `query`, and your `datasource` too. in fact, i changed it to `gabor`, and it still worked. strange.

Reply:
It's because the resource part is used when building the scope/target (e.g. `datasources:uid:{uid}`) the action applies to. But since you told me we don't check the scope there yet, it does not change a thing for now. Only in the future once you want to check if the user had `datasources:query` on `datasources:uid:ds1` it is going to matter. :slightly_smiling_face:

Reply:
Something to think about before we add any permission check here. Is the datasource resource going to be owned by `query.grafana.app` or is that going to be another api, something like `datasource.grafana.app`?

Before we commit to something here it is worth discussing that part first because of how authz is going to work.

If the datasource resource is owned by another api then permissions are going to be tied to that api but the permission to query that datasource is going to be tied to this api.

E.g. read access of a datasource would be `GET datasource.grafana.app/datasource/{name}` but for a user to query it (with how its setup now) it would require `POST query.grafana.app/query/{name}`

Reply:
So far, we have taken `datasources:read` to mean a user/service can query the datasource, querier service included. `query` isn't considered a write operation, best I know.

I am not sure we have used the word "owned" @kalleep, but query service has access on it.

When it comes to the `datasource.grafana.app`, there isn't a single one. Each datasource has its own subdomain value on top of it, so `prometheus.datasource.grafana.app`. Query service invokes requests on them too, with the same permission `datasources:read`.

Reply:
To give you more context we are working on replacing rbac for app platform where the backing engine is [openfga](https://github.com/openfga/openfga).

Then plan on how permissions are going to look is much more compatible with k8s. So you would use k8s verb to grant permissions to apis and their resources.

With this setup it would mean that a permissions that looks like `verb:get group:loki.datasources.grafana.app resource:datasources name:1` would grant a user access to use that api (also list that specific datasource). This would also mean that with how query service is currently setup we would need to grant `verb:create query.grafana.app resource:query name:1` in order to query that specific data source.

Reply:
@kalleep thanks for the extra info.

regarding the permissions.. some more context:
1. based on the "grafana" permissions, `datasources:read` and `datasources:query` are separate permissions. 
2. traditionally, to query a datasource meant sending a POST HTTP request to the `/api/ds/query` path, and there the `datasources:query` permission was checked.
3. we want to keep compatibility with the API from [2], so now the same POST HTTP request can be sent to `/apis/query.grafana.app/v0alpha1/namespaces/default/query`. this does not really map well to the kubernetes-style setup, i mean, we are not really creating a query, we are sending a POST HTTP request, where the request-body contains a JSON describing what we want to query, the query happens, and the HTTP response will contain the results. with that said, probably the "we CREATE a QUERY" is the closest we can describe it k8s-style.

with that said,do you think the current approach in this draft PR wrong (in that case, how would you recommend to change it?), or is it ok?
also, the plan to move to the `openfga` approach.. how far in the future is that?